README.md
# QingShuJia
[轻书架APP](https://github.com/cyb233/QingShuJia/releases/latest)

一款基于Fusion App制作的轻书架客户端

[轻书架官网](https://www.acgdmzy.com)

[爱发电赞助](https://afdian.net/@Schwi)


history.md
历史更新日志：
0.78
#增加设备信息查看（用于反馈）
#增加侧栏图片（P站ID 82024208）
0.77
#增加新页面支持
0.76
#修复“关于”页一项参数在安卓7.0以下报错的问题
0.75
#更新URL为远程获取
#部分文本修改
0.74
#优化更新信息获取，解决GitHubAPI(用于获取版本号和下载链接)在国内的访问问题
0.73
#优化各主页面标题显示
0.72
#修改加群按钮群号为在线获取，并改为列表形式以便添加修改
0.71
#修复书架按钮错误
0.70
#适配新的书架页面
#优化拦截音量键翻页仅在阅读页执行
#总感觉当初计划改啥来着但想不起来了
0.69
#移除了"屏蔽长按"
#增加推送型公告板
#好像还改了别的啥但想不起来了
0.68
#添加网站作者赞助(afdian.net)
#优化搜索
0.67
#尝试屏蔽选中功能预防闪退(onselectstart)
0.66
#修复了一些问题
0.65
#修复设置页部分设置不会立即生效的问题
#按钮调整
0.64
#修复一些小问题
0.63
#新增设置页(试行)
#新增设置项(屏蔽长按)
0.62
#屏蔽长按以避免闪退bug(存图请尝试菜单以浏览器打开)
0.61
#优化全屏阅读
0.60
#修复bug
0.59
#适配书架整理功能
0.58
#启动页和顶栏按钮分别设置
0.57
#自定义页针对顶栏按钮即时生效
#修复顶栏按钮bug
0.56
#增加支持自定义启动页（顶栏按钮共用此设置），对应设置位于“关于”页
0.55
#修改顶栏按钮链接至“最近更新”
0.54
#悬浮按钮形状改回圆形
#修改搜索框按钮颜色
#优化一些提示
0.53
#点击外链自动转浏览器打开，避免各种问题
#解除首页按钮的提示
#优化部分按钮判断
0.52
#优化WebView低版本判断提示(每次启动弹窗改为每日一次)
0.51
#修复翻页功能
#网站未实装按钮改为提示
0.50
#修复编辑按钮
0.49
#适配网站原生夜间模式
#修复一系列问题
0.48
#调整夜间模式网页配色
#修复夜间模式开关错误
#修复一处UA问题
0.47
#优化夜间模式
#增加上一页/下一页按钮
#按钮半透明化处理
#长按菜单回归
#优化访问统计
0.46
#优化夜间模式
#优化访问统计
0.45
#优化检查更新失败的显示
#尝试粗略统计用户量
0.44
#修复初次运行两个判断问题
#添加编辑按钮判断条件
0.43
#尝试性夜间模式
0.42
#支持了上架和编辑书籍功能
0.41
#修复退出全屏会返回章节目录的问题
#又双叒优化WebView版本判断
0.40
#优化WebView版本判断
#修复显示留白问题
0.39
#退出软件按钮移位
#增加章节按钮
#强化对低版本WebView的提示
0.38
#修复热门书籍按钮
0.37
#修复部分按钮功能
0.36
#修复全屏时切换界面会显示导航栏的问题
0.35
#添加判断WebView版本是否过低
#修复记住密码功能
0.34
#更换内核，修复了排版问题
#少量其他改动
#修改了UA
0.33
#修复值错误导致全屏按钮默认不显示的问题
0.32
#修改群号至新群号
0.31
#是否显示全屏按钮支持记住状态
#支持修改阅读字体大小
0.30
#历史记录换源至码云
0.29
#修复了一个网络问题的影响
0.28
#检查更新添加历史更新日志内容
0.27
#在“关于”中添加了轻书架用户协议
0.26
#阅读页添加了全屏按钮(返回键退出全屏)
#右上角可以开关按钮状态(仅当前启动有效)
0.25
#添加了“打开链接”功能(如果收到别人分享的链接也能打开了！)
#检查更新添加了个没啥意义的功能(当你的版本号比GitHub的还新时会提示，不过你们应该没机会看到)
0.24
#可选更新
#一些没啥影响的修改
0.23
#检查更新功能优化 #1 
0.22
#非实质性更新，仅是为了维护方便，可以不更
0.21
#增加启动软件时检查更新并提示 #1 
0.20
#修复了下载文件名错误的问题 #2 
0.19
#修复了更新说明的换行问题
#增加了更新时间
0.18
#添加了检查更新功能
0.17
-
0.16
-
0.15~
-


update-token.js
// ==UserScript==
// @name         轻书架更新token
// @namespace    Schwi
// @version      0.1
// @description  获取用户名和更新已保存的轻书架token
// @author       Schwi
// @exclude      https://www.acgdmzy.com*id
// @match        https://www.acgdmzy.com/*
// @grant        none
// ==/UserScript==


window.onload=function(){

    //展开头像
    if ( localStorage.getItem('token') != null ) {
        document.getElementsByClassName("v-avatar ml-1")[0].click();
    }
    //获取token和用户名，如果token不一致则更新
    setTimeout(function(){
        var username
        if ( localStorage.getItem('token') == null ) {
            username = null
        } else {
            username = document.getElementsByClassName("v-list-item__title")[7].innerText;
            sessionStorage.setItem('username',username);
            if ( localStorage.getItem('user_app_' + username) != null && localStorage.getItem('user_app_' + username) != localStorage.getItem('token')) {
                localStorage.setItem('user_app_' + username,localStorage.getItem('token'));
            }
        }
    },50);
    //收起头像
    setTimeout(function(){
        if ( localStorage.getItem('token') != null ) {
            document.getElementsByClassName("v-avatar ml-1")[0].click();
        }
    },100);

}


Multiple-accounts.user.js
// ==UserScript==
// @name         轻书架切换登录
// @namespace    Schwi
// @version      0.1
// @description  轻书架账号切换，在任意轻书架页面url后添加“/id”访问即可，需要当前浏览器打开期间至少打开过一次正常网站页面
// @author       Schwi
// @match        https://www.acgdmzy.com/*
// @grant        none
// ==/UserScript==

console.log('脚本启动');
var username;
var tokenGetFinish;
window.onload=function(){
    console.log('获取token和用户名，如果token不一致则更新');
    //展开头像
    if ( localStorage.getItem('token') != null ) {
        console.log('展开头像');
        document.getElementsByClassName("v-avatar ml-1")[0].click();
    }
    //获取token和用户名，如果token不一致则更新
    setTimeout(function(){
        if ( localStorage.getItem('token') == null ) {
            console.log('未获取到token，未登录');
            username = null;
        } else {
            username = document.getElementsByClassName("v-list-item__title")[7].innerText;
            sessionStorage.setItem('username',username);
            if ( localStorage.getItem('user_app_' + username) != null && localStorage.getItem('user_app_' + username) != localStorage.getItem('token')) {
                console.log('token不一致,更新token');
                localStorage.setItem('user_app_' + username,localStorage.getItem('token'));
            }
        }
        tokenGetFinish = true
        console.log('获取完毕');
    },50);
    //收起头像
    setTimeout(function(){
        if ( localStorage.getItem('token') != null ) {
            console.log('收起头像');
            document.getElementsByClassName("v-avatar ml-1")[0].click();
        }
    },100);
}
window.onhashchange=function(){
    console.log('页面变化：' + window.location.href);
    if ( /^.*\/id$/g.test(window.location.href) ){//判断url
        if (tokenGetFinish) {
            console.log('账号切换页');
            //获取token和用户名，如果token不一致则更新
            if ( localStorage.getItem('token') == null ) {
                console.log('未获取到token，未登录');
                username = null;
            } else {
                username = sessionStorage.getItem('username');
                console.log('已登录' + username);
                if ( localStorage.getItem('user_app_' + username) != null && localStorage.getItem('user_app_' + username) != localStorage.getItem('token')) {
                    console.log('更新token');
                    localStorage.setItem('user_app_' + username,localStorage.getItem('token'));
                }
            }
            //导入HTML
            setTimeout(function(){
                console.log('载入账号切换页')
                document.head.innerHTML='<meta charset="UTF-8"><meta name="viewport" content="width=device-width, initial-scale=1">';
                document.body.innerHTML='<p id="login" "center"></p><form><fieldset><legend align="center">请选择账号:</legend><div align="center"><input type="radio" id="id0" name="contact" value="id0"><label for="id0" id="id00">账号1</label></input><br><input type="radio" id="id1" name="contact" value="id1"><label for="id1" id="id11">账号2</label></input><br><input type="radio" id="id2" name="contact" value="id2"><label for="id2" id="id22">账号3</label></input></div></fieldset></form><div align="center"><button type="submit" onclick="localStorage.removeItem(localStorage.getItem(\'id0\'));localStorage.removeItem(localStorage.getItem(\'id1\'));localStorage.removeItem(localStorage.getItem(\'id2\'));localStorage.removeItem(\'id0\');localStorage.removeItem(\'id1\');localStorage.removeItem(\'id2\');document.getElementById(\'id00\').innerText = \'账号1\';document.getElementById(\'id11\').innerText = \'账号2\';document.getElementById(\'id22\').innerText = \'账号3\';alert(\'已清空！\');">清空记录</button><button type="submit" onclick="var item = null;var obj = document.getElementsByName(\'contact\');for (var i = 0; i < obj.length; i++) {if (obj[i].checked) {num = i;item = obj[i].value;obj[i].checked = false;}}if (item == null) {alert(\'请选择账号！\');} else {var token = localStorage.getItem(\'token\');if ( token == null ) {alert(\'您还没有登录账号！\');} else {localStorage.setItem(item,username);localStorage.setItem(\'user_app_\' + username,token);document.getElementById(item + num).innerText = username;alert(\'已保存用户：\' + username);}}">添加/修改</button><button type="submit" onclick="var item = null;var obj = document.getElementsByName(\'contact\');for (var i = 0; i < obj.length; i++) {if (obj[i].checked) {item = obj[i].value;obj[i].checked = false;}}if (item == null) {alert(\'请选择账号！\');} else {var username = localStorage.getItem(item);if ( username == null ) {alert(\'您还没有保存账号！\');} else {var token = localStorage.getItem(\'user_app_\' + username);localStorage.setItem(\'token\',token);alert(\'已切换用户：\' + username);window.location.href=\'\/#\/home\';}}">切换账号</button></div>';
            },100);
            //修改选项名
            setTimeout(function(){
                console.log('修改页面内容');
                if ( username != null ){
                    document.getElementById('login').innerText = '当前登录用户昵称为：' + username;
                } else {
                    document.getElementById('login').innerText = '当前未登录';
                }
                if ( localStorage.getItem('id0') != null ) {
                    document.getElementById('id00').innerText = localStorage.getItem('id0');
                }
                if ( localStorage.getItem('id1') != null ) {
                    document.getElementById('id11').innerText = localStorage.getItem('id1');
                }
                if ( localStorage.getItem('id2') != null ) {
                    document.getElementById('id22').innerText = localStorage.getItem('id2');
                }
            },150)
        } else {
            alert('请先访问正常页面')
            window.location.href='/#/home';
        }
    }
}
